packages/xo-server/src/vhd-merge.js

```diff
@@ -620,20 +627,23 @@ export class Vhd {
   }
 
   // Write a context footer. (At the end and beginning of a vhd file.)
-  async writeFooter () {
+  async writeFooter (onlyEndFooter = false) {
     const { footer } = this
 
-    const offset = this.getEndOfData()
     const rawFooter = fuFooter.pack(footer)
+    const eof = await this._handler.getSize(this._path)
+    // sometimes the file is longer than anticipated, we still need to put the footer at the end
```

Comment:
When can this happen?
Shoulnd't we truncate the file instead?

Reply:
1) I don't know exactly why some tools do that
2) I guess I'll do it during the compression.